Library/brew.sh

```diff
@@ -140,12 +173,47 @@ case "$HOMEBREW_COMMAND" in
   --config)    HOMEBREW_COMMAND="config";;
 esac
 
-if [[ -f "$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.sh" ]] ; then
+if [[ -f "$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.sh" ]]
+then
   HOMEBREW_BASH_COMMAND="$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.sh"
-elif [[ -n "$HOMEBREW_DEVELOPER" && -f "$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh" ]] ; then
+elif [[ -n "$HOMEBREW_DEVELOPER" && -f "$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh" ]]
+then
   HOMEBREW_BASH_COMMAND="$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh"
 fi
 
+if [[ "$HOMEBREW_COMMAND" = "help" || "$1" = "-h" || "$1" = "--help" ]]
+then
+
+  # Display sub command help for `brew help foo` and `brew foo --help`
+  if [[ "$HOMEBREW_ARG_COUNT" -gt 0 ]]
+  then
+    if [[ "$HOMEBREW_COMMAND" = "help" ]]
+    then
+      HOMEBREW_COMMAND="$HOMEBREW_LIBRARY/Homebrew/cmd/$1"
+    else
+      HOMEBREW_COMMAND="$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND"
+    fi
+
+    if [[ -f "$HOMEBREW_COMMAND.sh" ]]
+    then
+      HOMEBREW_COMMAND="$HOMEBREW_COMMAND.sh"
+    elif [[ -f "$HOMEBREW_COMMAND.rb" ]]
+    then
+      HOMEBREW_COMMAND="$HOMEBREW_COMMAND.rb"
+    else
+      echo "Unknown command \`$@\`"
+      echo
+      echo "$HOMEBREW_HELP"
+      exit 1
+    fi
+    grep "#:" "$HOMEBREW_COMMAND" | cut -c3- | tr -d \` | sed "s/^\* //"
+    exit 0
+  else
+    echo "$HOMEBREW_HELP"
+    exit 0
+  fi
+fi
```

Comment:
I'm :-1: to handle help flag on bash script. It's a duplication of the ruby version and fails to provide any advantage. In addition, it makes `brew.sh` more complicate than it used to be.

At the same time, I'm :-1: to use grep/cut to read help message. feel a little code smell to me.

Why cannot we handle all of these in ruby library and store help message using proper string rather than comments.